server.go

```diff
@@ -106,6 +113,10 @@ func NewServer(opt ...ServerOption) *Server {
 	for _, o := range opt {
 		o(&opts)
 	}
+	if opts.codec == nil {
+		// Set the default codec.
+		opts.codec = &protoCodec{}
```

Comment:
That's not how Go interfaces work. Here, `protoCodec` is an empty struct. It doesn't need to be a pointer, because the methods for the Codec interface are on `protoCodec`, not `*protoCodec`. If a user has a complex codec, they can implement those methods on `*myCodec` and pass `&myCodec{...}` to the relevant functions in this package.